src/compiler/checker.ts

```diff
@@ -13270,7 +13264,7 @@ namespace ts {
             return errorReported;
         }
 
-        function reportImplicitAnyError(declaration: Declaration, type: Type) {
+        function reportImplicitAnyError(noImplicitAny: boolean, declaration: Declaration, type: Type) {

```

Comment:
May be rename this to `reportImplicitAnyErrorOrSuggestion` instead?

Reply:
It's not necessary to take `noImplicitAny` as a parameter since that's already in scope. In every instance of calling this the argument is `noImplicitAny`; in one case the argument is `true` but it's within `if (noImplicitAny)`.

Reply:
I chose to shorten it to `reportImplicitAny`

Reply:
@andy-ms good catch. I'll get rid of the parameter.